src/track/beatutils.cpp

```diff
@@ -289,16 +289,22 @@ double BeatUtils::makeConstBpm(
 
     // Create a const region region form the first beat of the first region to the last beat of the last region.
 
-    const double minRoundBpm = 60 * sampleRate / longestRegionBeatLengthMax;
-    const double maxRoundBpm = 60 * sampleRate / longestRegionBeatLengthMin;
-    const double centerBpm = 60 * sampleRate / longestRegionBeatLength;
+    const double minRoundBpm = 60.0 * sampleRate / longestRegionBeatLengthMax;
+    const double maxRoundBpm = 60.0 * sampleRate / longestRegionBeatLengthMin;
+    const double centerBpm = 60.0 * sampleRate / longestRegionBeatLength;
 
     //qDebug() << "minRoundBpm" << minRoundBpm;
     //qDebug() << "maxRoundBpm" << maxRoundBpm;
     const double roundBpm = roundBpmWithinRange(minRoundBpm, centerBpm, maxRoundBpm);
 
     if (pFirstBeat) {
+        // Move the first beat as close to the start of the track as we can. This is
+        // a constant beatgrid so "first beat" only affects the anchor point where
+        // bpm adjustments are made.
         *pFirstBeat = constantRegions[startRegionIndex].firstBeat;
+        const double roundedBeatLength = 60.0 * sampleRate / roundBpm;
+        const double beatsBeforeFirst = std::floor(*pFirstBeat / roundedBeatLength);
+        *pFirstBeat -= (roundedBeatLength * beatsBeforeFirst);
```

Comment:
We have the same code in double BeatUtils::adjustPhase() as shortened:  
```
    const double beatLength = 60 * sampleRate / bpm;
    const double startOffset = fmod(firstBeat, beatLength);
```
Can you adopt it here? 

For now it would be save to move it here. 
However i think we should keep it to both places makes the functions more robust against future refactoring where the fist beat is the first down beat.

Reply:
Did you consider adopting the shorter code here as well?

Reply:
sorry I missed this, updated -- I think correctly?